### PR TITLE
cmdstan: add aarch64-darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Following #203381.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build cmdstan cmdstan.passthru.tests

File: pkgs/development/compilers/cmdstan/default.nix [hunk 1]
```diff
@@ -38,6 +31,11 @@ stdenv.mkDerivation rec {
       src/test/interface/stansummary_test.cpp
   '';
 
+  preConfigure = ''
+    mkdir -p bin
+    ln -s ${stanc}/bin/stanc bin/stanc
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/cmdstan/default.nix:36]: Is this really required? Can't we remove the static path instead?

**Comment by wegank** [comment]
The build mysteriously fails on my aarch64-darwin machine...

@ofborg build cmdstan cmdstan.passthru.tests

File: pkgs/development/compilers/cmdstan/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/compilers/cmdstan/default.nix:36]: The makefile [requires that](https://github.com/stan-dev/cmdstan/blob/develop/make/stanc#L47) an executable to be at `bin/stanc`, and patching `bin/stanc$(EXE)` around gave me errors on tests. I guess it's easier to leave it as is.

Comment (SuperSandro2000) [reply pkgs/development/compilers/cmdstan/default.nix:36]:
```suggestion
    ln -s ${buildPackages.stanc}/bin/stanc bin/stanc
```
then I think we want to do this to make it cross friendly

Comment (wegank) [reply pkgs/development/compilers/cmdstan/default.nix]: Does. Sorry to have to rebase it for storage constraints.

**Pull request merged**